### PR TITLE
CLI base

### PR DESCRIPTION
**Pull request by mixmix**
This PR adds:
- **compling** (ts => js)
    - I have copied the sdk setup 
    - I wasted time trying something new. we should come back and update this when we figure out a beautiful TS pattern
- **commander**, which captures CLI input and will allow us to add programmatic functions nicely
- deletes some unused `questions.ts` file

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,13 +45,11 @@
...
     "ansi-colors": "^4.1.3",
     "cli-progress": "^3.12.0",
+    "commander": "^12.0.0",
     "debug": "^4.3.4",
     "dotenv": "^16.4.1",
     "ethers": "^5.7.2",
     "inquirer": "8.0.0",
-    "ts-node": "^10.9.2",
-    "typescript": "^4.8.4",
-    "viem": "^2.7.8",
```
Comment (mixmix) [inline package.json:44]: `viem` was causing problems with compiling. Not even used atm

**Review by rh0delta — APPROVED** [review]
this looks good to me, @frankiebee do you want to give it a once over too?

File: package.json [hunk 2]
```diff
@@ -2,10 +2,11 @@
   "name": "@entropyxyz/cli",
   "version": "1.0.0",
   "description": "",
+  "type": "module",
```
Comment (frankiebee) [inline package.json:5]: is this needed?

File: package.json [hunk 3]
```diff
@@ -17,11 +18,20 @@
...
+  },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/entropyxyz/cli.git"
   },
-  "keywords": [],
+  "keywords": [
+    "entropy",
+    "entropyxyz"
+  ],
```
Comment (frankiebee) [inline package.json:34]: lets add `entropy protocol` here

File: src/common/ascii.ts [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
+  @@@@@      @@@@@ @@@@ @@@@@   @@@@@       @@@@ @@@@ @@@@@ @@@@ @@@@@ @@@@@
+  @@@@@ @@@@ @@@@@ @@@@ @@@@@   @@@@@       @@@@ @@@@ @@@@@ @@@@ @@@@@ @@@@@
+  @@@@@ @@@@ @@@@@ @@@@ @@@@@   @@@@@       @@@@ @@@@ @@@@@ @@@@ @@@@@ @@@@@
+  @@@@@ @@@@ @@@@@ @@@@ @@@@@   @@@@@       @@@@ @@@@ @@@@@ @@@@ @@@@@ @@@@@
+  @@@@@@@@@@ @@@@@ @@@@ @@@@@@@ @@@@@       @@@@@@@@@ @@@@@@@@@@ @@@@@@@@@@@
+  @@@@@@@@@@ @@@@@ @@@@ @@@@@@@ @@@@@       @@@@@@@@@ @@@@@@@@@@ @@@@@@@@@@@
+                                                      @@@@@@            TEST
+                                                      @@@@@@            *NET
+                                                      @@@@@@     ENTROPY-CLI
+                                                      @@@@@@     COREv0.0.12
```
Comment (frankiebee) [inline src/common/ascii.ts:20]: this is incorrect should be `0.1.0`

Comment (frankiebee) [reply src/common/ascii.ts:20]: side note for sdk: make this exportable?

File: tsconfig.json [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
-    "moduleResolution": "node",
+    "incremental": true,
     "typeRoots": [
       "node_modules/@types",
       "@types/",
-      "@entropyxyz/sdk",
-      "@entropyxyz/sdk/keys"
-    ]
+      "@polkadot/types/*",
+      "substrate-generated-types/interfaces"
```
Comment (frankiebee) [inline tsconfig.json:11]: not needed

```suggestion
```

File: tsconfig.json [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
-  "include": [
-    "src/**/*.ts"
-  ],
   "exclude": [
-    "node_modules"
+    "node_modules",
+    "dist",
+    "build",
+    "examples",
+    "old",
```
Comment (frankiebee) [inline tsconfig.json:22]: not needed
```suggestion
```

File: tsconfig.json [hunk 7]
```diff
@@ -1,28 +1,28 @@
...
+    "substrate-generated-types"
```
Comment (frankiebee) [inline tsconfig.json:23]: not needed

```suggestion
```

**Review by frankiebee — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (mixmix) [reply package.json:5]: TODO: later :eyes:

File: src/common/ascii.ts (hunk 4 above)
Comment (rh0delta) [reply src/common/ascii.ts:20]: The versioning for core is fixed in a branch that I have, it's one of the issues I took care of this morning

**Comment by mixmix** [comment]
@frankiebee this is ready for sign off. This Changes Requested :lock: is both good and annoying :japanese_ogre:

File: README.md [hunk 8]
```diff
@@ -1,8 +1,26 @@
 # Entropy CLI
 
+> :warning: This tool is early development. As such, a lot of things do not work. Feel free to play around with it and report any issues at [github.com/entropyxyz/cli](https://github.com/entropyxyz/cli).
```
Comment (rh0delta) [inline README.md:3]:
```suggestion
> :warning: This tool is in early development. As such, a lot of things do not work. Feel free to play around with it and report any issues at [github.com/entropyxyz/cli](https://github.com/entropyxyz/cli).
```

**Review by rh0delta — APPROVED** [review]
minor change in readme, otherwise it looks good

**Pull request merged**